### ISSUE TITLE
comboBox.setItems(*) is removing comboBox value

### ISSUE THREAD

**Issue by utrucceh** [vaadin/framework#9114]
for example:
comboBox.setValue(x);
comboBox.setItems(y); 
comboBox.getValue() ==> Should return "x", but returning "null"

**Comment by ssindelar** [comment]
setItems overwrites the existing items. x is no longer an item of the combo box, so I think the behaviour getValue is correct. 

But I think this changed from Vaadin 7, because when I remember correctly at least for some time you could remove the selected value and getValue() would still return the removed value.

**Comment by utrucceh** [comment]
But there is something:
-after setItems(y), x is staying on View, if getValue become null, why it is staying
-sometimes we are replacing items for user, and there is x in new items, also x is removing

**Comment by christoph-frick** [comment]
I'd like to second that part about the "visual cue for the user sticks" after the `setItems` part.  Here is an example, that shows that problem:

```groovy
def cb = new ComboBox()
cb.items = [1]
cb.value = 1
setContent( 
	new CssLayout(
		cb, 
		new Button("change items and set value=null in CB; but visual value in CB remains 1", {
			cb.items = [42]
			cb.value = null
		} as Button.ClickListener),
	)
)
```

[Full springboot example](https://gist.github.com/christoph-frick/19558fd6b8e2d7d2f52d0b32514bc270)

Once the button gets clicked, the items in the CB change, but the `1` still is shown like selected.  Expected behaviour would be to have `null` (nothing) selected.

**Comment by hesara** [comment]
Confirmed with current master (post-8.1.0.beta1), Java version of the snippet with some additional comments on what happens
```
final ComboBox<Integer> combo = new ComboBox<>();
combo.setItems(1);
addComponent(combo);
addComponent(new Button("This should clear the displayed value", e -> {
    // the shown value is visually cleared only if setValue(null) is done first, though the value is set to null in either case
    combo.setItems(42);
    combo.setValue(null);
}));
// this will print "null" even in the scenario where the user does
// - select 1 from combo
// - click the first button, which clears the internal value but leaves "1" shown
// - click the second button
addComponent(new Button("Show it", e -> System.out.println(combo.getValue())));
```

**Comment by TatuLund** [comment]
The visual problem is as the same bug as reported in 
https://github.com/vaadin/framework/issues/11454 and fixed in Vaadin 8.7.3
